lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -1942,7 +1942,12 @@ ADContext::createPrimalValueStruct(const DifferentiationTask *task,
     pvStruct->setAccess(AccessLevel::FilePrivate);
     break;
   default:
-    llvm_unreachable("The original function cannot have external linkage");
+    // When the original function has external linkage, we create an internal
+    // struct for use by our own module. This is neccessary for cross-cell
+    // differentiation in Jupyter.
+    // TODO: Add a test in the compiler that exercises a similar situation as
+    // cross-cell differentiation in Jupyter.
+    pvStruct->setAccess(AccessLevel::Internal);
```

Comment:
The function is loaded at the point that this case gets triggered, but the linkage is still public_external. Here's a dump of the function at this line:
```
// squared(_:)
sil public_external [differentiable source 0 wrt 0 jvp @AD__$s14__lldb_expr_157squaredyS2fF__jvp_src_0_wrt_0] @$s14__lldb_expr_157squaredyS2fF : $@convention(thin) (Float) -> Float {
// %0                                             // users: %3, %3
bb0(%0 : $Float):
  %1 = metatype $@thin Float.Type                 // user: %3
  // function_ref static Float.* infix(_:_:)
  %2 = function_ref @$sSf1moiyS2f_SftFZ : $@convention(method) (Float, Float, @thin Float.Type) -> Float // user: %3
  %3 = apply %2(%0, %0, %1) : $@convention(method) (Float, Float, @thin Float.Type) -> Float // user: %4
  return %3 : $Float                              // id: %4
} // end sil function '$s14__lldb_expr_157squaredyS2fF'
```

Reply:
Ok! It's better to also add a `@usableFromInline` so that we can do higher-order cross-cell differentiation in the future, but i'll add it later.